src/EventCalendar.js

```diff
@@ -65,120 +104,526 @@ class EventCalendar {
+     */
+    styleRepeatOn (target) {
+        let _self = this,
+            $elem = _self.clndr.element.find('[data-day="' + target.format(_self.dateFormat) + '"]');
+
+        $elem.attr('aria-label', 'Event will repeat on ' + target.format(_self.dateFormatLong) + ' based on the chosen repeat pattern')
+             .parent()
+             .addClass('event-repeat');
+    }
+
+    /**
+     * Remove styling from a date and reset it to the 'neutral' state
+     * 
+     * @param {moment} target The date to style
+     */
+    styleClear (target) {
+        let _self = this,
+            $elem = _self.clndr.element.find('[data-day="' + target.format(_self.dateFormat) + '"]');
+
+        $elem.attr('aria-label', target.format(_self.dateFormatLong) + '.')
+             .parent()
+             .removeClass('event-add event-del event-repeat');
+
+        // Announce the new status through a live region (instead of the previous status)
+        _self.updateLiveRegion('Unselected. Event will not repeat on ' + target.format(_self.dateFormatLong));
+    }
+
+    /**
+     * Restore all dates to their initial state (when the calendar was initialised)
+     */
+    styleClearAll (target) {
+        let _self = this,
+            $elems = _self.clndr.element.find('.day-contents');
+
+            $elems.each(function() {
+                let $elem = $(this),
+                    $elemParent = $elem.parent(),
+                    date = moment($elem.data('day')).format(_self.dateFormatLong),
+                    ariaLabel = 'Unselected';
+
+                // If event was passed in the `events` array, it will be reset back to `.selected` and needs labelling
+                if ($elemParent.attr('class').indexOf('event') >= 0) {
+                    ariaLabel = 'Selected. Event will repeat on this day';
+                }
+
+                $elem.attr('aria-label',  date + '. ' + ariaLabel);
+                $elemParent.removeClass('event-add event-del event-repeat');
+            });
+        }
+
+    /**
+     * Check whether `date` is in the same month as `dateToCompareTo`
+     * 
+     * @param {moment} dateToCompareTo Haystack
+     * @param {moment} date Needle
+     */
     static isInMonth (dateToCompareTo, date) {
-        return date.date.date.month() === dateToCompareTo.month();
+        return date.month() === dateToCompareTo.month();
     }
     
+    /**
+     * Check whether `date` is in the same year as `dateToCompareTo`
+     * 
+     * @param {moment} dateToCompareTo Haystack
+     * @param {moment} date Needle
+     */
     static isInYear (dateToCompareTo, date) {
-        return date.date.date.year() === dateToCompareTo.year();
+        return date.year() === dateToCompareTo.year();
+    }
+
+    /**
+     * Check whether two moment instances are the same date
+     * 
+     * @param {moment} a Haystack
+     * @param {moment} b Needle
+     */
+    static matchDates (a, b) {
+        return a._i != b._i;
```

Comment:
Should we be using `a.isSame(b)` here to leverage moment's public API for determining if 2 dates are the same. Might be nicer than using this pseudo private `_i` property, which looks as though it references the builtin `Date` object.

Although I may have the wrong end of the stick because this reads as though `matchDates` returns `true` if the date objects used to construct the moment object are not the same instance of `Date`, therefore 2 `Date` objects that have the same date would not be equal 🤔

Reply:
💥

Reply:
can change to `return a.format(this.dateFormat) != b.format(this.dateFormat);` which seems to be a more appropriate way rather than referencing the internals

Reply:
Sounds good to me, although still feels as though the method name is misleading, as we're asserting in the expression that they do not match but it's called `matchDates` if I've not misunderstood